snuba/datasets/configuration/loader.py

```diff
@@ -19,3 +22,16 @@ def load_configuration_data(
     assert isinstance(config, dict)
     validate(config, validation_schemas[config["kind"]])
     return config
+
+
+def load_config_built_storage_keys() -> dict[str, str]:
+    """
+    Loads all storage keys from configuration files
+    """
+    return {
+        key.upper(): key
+        for key in [
+            safe_load(open(config_file))["storage"]["key"]
```

Comment:
I'm still not a fan of this, I think there should be one single entrypoint for any loading of any storage configuration file.

Reply:
The key dictionary could be moved into the same place as the storage factory. Then it could be populated at the same time the storages are loaded from the config.

Reply:
Yeah, that sounds like a good approach to me.

Reply:
I don't think we can move this dictionary out into the Storage factory file and then import it in the init file where the enum exists. This is the first thing I tried and it caused a circular import since the storage factory file imports the enum itself.